jetbrains-core/src/software/aws/toolkits/jetbrains/utils/ResourceOperationAgainstCodePipeline.kt

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.utils
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.ui.showYesNoDialog
+import software.amazon.awssdk.services.resourcegroupstaggingapi.ResourceGroupsTaggingApiClient
+import software.amazon.awssdk.services.resourcegroupstaggingapi.model.GetResourcesRequest
+import software.amazon.awssdk.services.resourcegroupstaggingapi.model.TagFilter
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.resources.message
+
+const val CODEPIPELINE_SYSTEM_TAG_KEY = "aws:codepipeline:pipelineArn"
+
+/**
+ * @property tagFilter used by ResourceGroupsTaggingApi to filter response by ResourceType (resourceTypeFilter)
+ */
+enum class ResourceType(val value: String, val tagFilter: String) {
```

Comment:
This seems like a very generic name for something so specialized.

Reply:
I'll rename it to TaggingResourceType.